WooCommerce/Classes/Tools/InfiniteScroll/ScrollWatcher.swift

```diff
@@ -0,0 +1,43 @@
+import UIKit
+
+/// Watches for scroll content offset changes and notifies its subscribers if the scroll position is over a threshold.
+///
+final class ScrollWatcher {
+    /// Emits a stream of scroll position percentages from 0.0 (the beginning) to 1.0 (the end) if over a given threshold whenever the user scrolls a
+    /// `UIScrollView` or subclass.
+    var scrollTrigger: Observable<Double> {
+        scrollTriggerSubject
+    }
+    private let scrollTriggerSubject: PublishSubject<Double>
+    private let scrollPositionThreshold: Double
+
+    private var offsetObservation: NSKeyValueObservation?
+
+    /// - Parameter scrollPositionThreshold: the threshold of scroll position percentage (from 0.0 to 1.0) to trigger an event.
+    init(scrollPositionThreshold: Double = 0.7) {
```

Comment:
Is there a reason on why we init it with `0.7` by default?
I suggest to not set a default value unless there is a reason behind it 👍 
```suggestion
    init(scrollPositionThreshold: Double) {
```

Reply:
I thought most of the table views would have the same trigger for fetching the next page, that's why I set 70% as a guess and we can tweak this default value later too. do you think each paginated list might want to have their own threshold?

Reply:
It makes sense, thanks for clarifying it 👍